tests/phpunit/TemplateTest.php

```diff
@@ -564,6 +564,30 @@ public function testConvertJournalToBook() {
        $this->assertEquals('cite book', $expanded->wikiname());
    }
     
+   public function testGadgetAPI() {
```

Comment:
What we need to test, if anything, is that nothing is echoed when gadgetapi.php is run (except the final output).  As you say, the functionality itself is tested in TemplateTest.php and elsewhere.  It's not clear that this tests that.